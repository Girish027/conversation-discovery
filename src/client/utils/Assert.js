/**
 * 24/7 Customer, Inc. Confidential, Do Not Distribute. This is an
 * unpublished, proprietary work which is fully protected under
 * copyright law. This code may only be used pursuant to a valid
 * license from 24/7 Customer, Inc.
 */

/**
 * Assertions module
 * @module helpers/assert
 */

const RX_EMPTY = /^\s*$/;

/**
 * Callback invoked when an assertion fails.
 * The default implementation throws an exception that includes a message describing the failed assertion.
 * @func _onFailure
 * @private
 * @memberOf module:helpers/assert
 * @param {String} msg Message to be shown when exception is thrown
 */
function onFailure(msg) {
  throw new Error(msg);
}

class Assert {
  /**
   * Assert a condition and throw an error if the condition is equivalent to `false`.
   * @func assert
   * @public
   * @memberOf module:helpers/assert
   * @param {Boolean} cond A condition to evaluate.
   * @param {String} [msg] The error message to throw.
   */
  static assert(cond, msg = 'assert failed') {
    if (!cond) {
      onFailure(msg);
    }
  }

  /**
   * Assert the given value to be an object.
   * @func isObject
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} obj An object to evaluate.
   * @param {String} [name] The name of the object.
   */
  static isObject(obj, name) {
    const msg = (name ? `${name} is not an object` : 'object expected');
    Assert.assert(typeof obj === 'object' && obj !== null, msg);
  }

  /**
   * Assert the given value to be undefined or null.
   * @func isUndefinedOrNull
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} obj An object to evaluate.
   * @param {String} [name] The name of the object.
   */
  static isUndefinedOrNull(obj, name) {
    const msg = (name ? `${name} is not undefined or null` : 'undefined or null expected');
    Assert.assert(obj === undefined || obj === null, msg);
  }

  /**
   * Assert the given value to be not undefined or null.
   * @func isNotUndefinedOrNull
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} obj An object to evaluate.
   * @param {String} [name] The name of the object.
   */
  static isNotUndefinedOrNull(obj, name) {
    const msg = (name ? `${name} is undefined or null` : 'undefined or null not expected');
    Assert.assert(obj !== undefined && obj !== null, msg);
  }

  /**
   * Assert the given value to be a boolean.
   * @func isBoolean
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} bool A boolean to evaluate.
   * @param {String} [name] The name of the boolean.
   */
  static isBoolean(bool, name) {
    const msg = (name ? `${name} is not a boolean` : 'boolean expected');
    Assert.assert(typeof bool === 'boolean', msg);
  }

  /**
   * Assert the given value to be a function.
   * @func isFunction
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} func A function to evaluate.
   * @param {String} [name] The name of the function.
   */
  static isFunction(func, name) {
    const msg = (name ? `${name} is not a function` : 'function expected');
    Assert.assert(typeof func === 'function', msg);
  }

  /**
   * Assert the given value to be a number.
   * @func isNumber
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} num A number to validate.
   * @param {String} [name] The name of the variable that stores the number.
   */
  static isNumber(num, name) {
    const msg = (name ? `${name} is not a number` : 'number expected');
    Assert.assert(typeof num === 'number', msg);
  }

  /**
   * Assert the given value is a string and not empty.
   * @func notEmpty
   * @public
   * @memberOf module:helpers/assert
   * @param {String} str A string to evaluate.
   * @param {String} [name] The name of the string.
   */
  static notEmpty(str, name) {
    const msg = (name ? `${name} is not a string or is empty` : 'non-empty string expected');
    Assert.assert(typeof str === 'string' && !RX_EMPTY.test(str), msg);
  }

  /**
   * Assert the given object has the given constructor.
   * @func instanceOf
   * @public
   * @memberOf module:helpers/assert
   * @param {Object} obj An object to evaluate.
   * @param {Object.constructor} constructor A constructor to evaluate.
   * @param {String} [className] The name of the class/constructor.
   */
  static instanceOf(obj, constructor, className) {
    const msg = (className ? `expected instance of ${className}` : 'unexpected instance type');
    Assert.assert(obj instanceof constructor, msg);
  }

  /**
   * Assert the given value to be a non-empty string, a number, or a boolean.
   * @func isScalar
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} value A variant to evaluate.
   * @param {String} [name] The name of the variable that stores the variant.
   */
  static isScalar(value, name) {
    const vType = typeof value;
    const msg = `${name ? `${name} ` : ''}must be string, number, or boolean`;
    Assert.assert(vType === 'string' || vType === 'number' || vType === 'boolean', msg);
    if (vType === 'string') {
      Assert.notEmpty(value, name);
    }
  }

  /**
   * Assert the given value to be an array.
   * @func isArray
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} value A variant to evaluate.
   * @param {String} [name] The name of the variable that stores the variant.
   */
  static isArray(value, name) {
    const msg = (name ? `${name} is not an array` : 'array expected');
    Assert.assert(Object.prototype.toString.call(value) === '[object Array]', msg);
  }
}


export default Assert;
