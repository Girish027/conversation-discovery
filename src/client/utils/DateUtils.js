import dateFormat from 'dateformat';

export const getDateFromTimestamp = (timestamp, locale = 'en-US') => new Date(parseInt(timestamp, 10))
  .toLocaleDateString(locale, {
    day: 'numeric',
    month: 'short',
    year: 'numeric',
  });

export const formatTimestamp = (timestamp, format = 'mmm d, h:MM TT') => {
  const dateObj = new Date(parseInt(timestamp, 10));
  return dateFormat(dateObj, format);
};
