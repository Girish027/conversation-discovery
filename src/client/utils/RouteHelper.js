import { basePath } from '../swagger.json';

const baseRoute = `${basePath}/ui`;

const routeParams = ({
  clientId, appId, accountId, projectId, runId, clusterId, componentClientId
}) => {
  let params = `?clientid=${clientId}&accountid=${accountId}&appid=${appId}&componentClientId=${componentClientId}`;
  if (projectId) {
    params += `&project=${projectId}`;
    if (runId) {
      params += `&run=${runId}`;
      if (clusterId) {
        params += `&cluster=${clusterId}`;
      }
    }
  }

  return params;
};

export const routeNames = {
  BASE_ROUTE: baseRoute,
  HOME_PAGE: basePath,
  DISCOVER_INTENTS: `${baseRoute}/discover-intents`,
  DISTRIBUTION_GRAPH: `${baseRoute}/run-overview`,
  TOPIC_REVIEW: `${baseRoute}/topic-review`
};

export const getRoute = (routeName, params = {}) => routeName + routeParams(params);
