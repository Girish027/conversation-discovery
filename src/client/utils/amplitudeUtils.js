// import amplitude from 'amplitude-js/amplitude';

export const initializeAnalytics = (analyticsKey, userName = '') => {
  // const analytics = amplitude.getInstance();
  // analytics.init(analyticsKey, userName);
};

export const logAmplitudeEvent = (eventName, eventSpecificData = {}) => {
  // amplitude.getInstance().logEvent(eventName, {
  //   ...eventSpecificData,
  // });
};
