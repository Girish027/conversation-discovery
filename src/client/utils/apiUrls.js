import { basePath } from '../swagger.json';

const root = (process.env.NODE_ENV === 'production'
  ? '//'.concat(window.location.host)
  : '//localhost:8080')
  .concat(basePath);

const clientInfoPath = (data) => `${root}/clients/${data.componentClientId}/accounts/${data.accountId}/applications/${data.appId}`;
const projectInfo = ({ projectId }) => `projects/${projectId}`;
// TODO remove mockServer once APIs are implemented
// const runsInfo = ({ runId }) => `runs/${runId}/mockServer/`;
const runsInfo = ({ runId }) => `runs/${runId}`;
const clusterInfo = ({ clusterId }) => `clusters/${clusterId}`;
const transcriptInfo = ({ transcriptId }) => `conversations/${transcriptId}`;
const conversationTranscriptInfo = (transcriptId) => `conversations?tId=${transcriptId}`;

// TODO: Curation
// API: ....conversations?transcriptIds=['abc', 'def']&moveTo=<new cluster id>

export const apiUrls = {
  authentications: `${root}/authentications?self=true`,
  userConfigUrl: `${root}/user/config`,
  conversationsConfigUrl: `${root}/user/conversationsConfig`,
  answersConfigUrl: `${root}/user/answersConfig`,
  downloadTemplate: (data) => `${clientInfoPath(data)}/templates`,
  projects: (data) => `${clientInfoPath(data)}/projects`,
  oneProject: (data) => `${clientInfoPath(data)}/${projectInfo(data)}`,
  runs: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/runs`,
  oneRun: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}`,
  downloadRun: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}/results?type=original`,
  clusters: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}/clusters`,
  oneCluster: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}/${clusterInfo(data)}`,
  conversations: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}/${clusterInfo(data)}/conversations`,
  transcript: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}/${clusterInfo(data)}/${transcriptInfo(data)}`,
  intents: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/runs/${data.runId}/${clusterInfo(data)}/intents`,
  conversationTranscript: (data, transcriptId) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}/${clusterInfo(data)}/${conversationTranscriptInfo(transcriptId)}`,
  interfaces: (data) => `${root}/clients/${data.componentClientId}/interfaces`,
  faqs: (data, interfaceId) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}/${clusterInfo(data)}/interfaceId/${interfaceId}/addToAnswers`
};
