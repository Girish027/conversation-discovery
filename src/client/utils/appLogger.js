// log to console if method level meets log level
function appLogger(options) {
  // indexes of method names match values of levels properties
  const loggerMethods = ['trace', 'debug', 'info', 'warn', 'error'];
  const levels = {
    TRACE: 0,
    DEBUG: 1,
    INFO: 2,
    WARN: 3,
    ERROR: 4,
    SILENT: 5
  };
  const envLevels = {
    development: levels.TRACE,
    production: levels.SILENT,
    test: levels.ERROR
  };

  // environment variables can override default log level
  let logLevel = levels.INFO;
  if (typeof process.env.LOG_LEVEL !== 'undefined') {
    logLevel = levels[process.env.LOG_LEVEL];
  } else if (typeof process.env.NODE_ENV !== 'undefined') {
    logLevel = envLevels[process.env.NODE_ENV];
  }

  const logger = {};
  loggerMethods.forEach((methodName, methodLevel) => {
    logger[methodName] = (...args) => {
      // only print out if method level is not less than log level
      if (methodLevel >= logLevel) {
        let msgs = [];
        if (options && options.tag) {
          msgs.push(options.tag);
        }
        msgs = msgs.concat(args);
        /* eslint-disable no-alert, no-console */
        console.log(...msgs);
      }
    };
  });
  return logger;
}

export default appLogger;
