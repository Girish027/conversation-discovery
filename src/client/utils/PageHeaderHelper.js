import { getRoute, routeNames } from 'utils/RouteHelper';
import { push } from 'connected-react-router';
import ObjectUtils from './ObjectUtils';
import Constants from '../constants';

// Note: order of items is the order Analyse Run breadcrumbs is rendered
const items = {
  discoverIntents: {
    label: 'discoverIntents',
  },
  projectView: {
    label: 'projectView',
    route: routeNames.DISCOVER_INTENTS
  },
  runView: {
    label: 'runView',
    route: routeNames.DISTRIBUTION_GRAPH
  },
};

const distributionItems = ObjectUtils.filterSpecificKeys(items, ['discoverIntents', 'projectView']);

export const goToBotOverview = () => {
  window.location.assign('#');
};
const getBreadCrumbItem = (label, data = {}, route, type = '', givenOnClick, readOnly = false) => {
  if (label) {
    const { dispatch, ...params } = data;
    let onClick;
    if (!readOnly) {
      if (givenOnClick) {
        onClick = givenOnClick;
      } else {
        onClick = () => dispatch(push(getRoute(route, params)));
      }
    }
    return {
      value: type + label,
      label,
      readOnly,
      onClick,
      ...params,
    };
  }
  return undefined;
};

const discoverIntents = 'Discover Intents';
const botOverview = 'Intent Discovery';
const overview = 'Overview';

const breadcrumbsMap = {
  botOverview: (data) => getBreadCrumbItem(botOverview, data, undefined, '', () => goToBotOverview()),
  discoverIntents: () => getBreadCrumbItem(discoverIntents, undefined, undefined, '', undefined, true),
  projectView: (data, route) => {
    let name = data.projectName;
    if (data.projectType && data.projectType === Constants.projects.SYSTEM) {
      name = Constants.LIVE_DATA_ANALYSIS;
    }
    return getBreadCrumbItem(name, data, route, 'project');
  },
  runView: (data, route) => getBreadCrumbItem(`${data.runName} ${overview}`, data, route, 'run'),
  clusterView: (data, route) => getBreadCrumbItem(data.clusterName, data, route, 'cluster'),
};

const defaultDetails = {
  title: () => discoverIntents
};

// Note: If action Items are to be added, it should be added in  detailsMap
// KeyName should be 'actionItems' .
// Ref: PageHeader component from '@tfs/ui-components'.
export const detailsMap = {
  [routeNames.BASE_ROUTE]: defaultDetails,
  [routeNames.HOME_PAGE]: defaultDetails,
  [routeNames.DISCOVER_INTENTS]: defaultDetails,
  [routeNames.TOPIC_REVIEW]: {
    title: () => 'Topic Review',
    breadcrumbItems: Object.values(items)
  },
  [routeNames.DISTRIBUTION_GRAPH]: {
    title: (runName) => `${runName} ${overview}`,
    breadcrumbItems: Object.values(distributionItems)
  },
};

/**
 * gets the Page Subheader title, breadcrumbs, action Items based on routeName
 * @param  {Object} data - should contain the following required params
 *  - required params
 *     - itsBaseURL, routeName, clientId, accountId, appId, dispatch
 *  - Optional combination params
 *    - projectName + projectId
 *    - runName + runId
 *    - clusterName + clusterId
 * @return {[Object]} should contain title of page, breadcrumbs array as expected by TFS UI PageHeader, actionItems
 */
export const getPageDetails = (data) => {
  let { routeName } = data;
  const { runName } = data;
  if (routeName[routeName.length - 1] === '/') {
    routeName = routeName.slice(0, routeName.length - 1);
  }
  const { [routeName]: detailsObj = {} } = detailsMap;
  const { title = () => {}, breadcrumbItems = [] } = detailsObj;
  const breadcrumbArray = [];
  breadcrumbItems.forEach((item) => {
    const breadcrumb = breadcrumbsMap[item.label](data, item.route);
    if (breadcrumb) {
      breadcrumbArray.push(breadcrumb);
    }
  });
  return {
    title: title(runName),
    breadcrumb: breadcrumbArray
  };
};
