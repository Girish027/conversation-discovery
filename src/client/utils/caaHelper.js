import appLogger from 'utils/appLogger';
import memoizeOne from 'memoize-one';

const devlog = appLogger({ tag: 'caaHelper' });

export const getSortedClientsList = (clientsList = []) => clientsList
  .map((config) => ([config.clientId.toLowerCase(), config]))
// eslint-disable-next-line no-nested-ternary
  .sort((a, b) => (a[0] > b[0] ? 1 : a[0] < b[0] ? -1 : 0))
  .map(([, config]) => config);


export const getDefaultCAA = (sortedClientsList) => {
  const { clientId, apps = [{}], componentClientId } = sortedClientsList[0];
  const { appId, accountId } = apps[0];
  return {
    clientId,
    appId,
    accountId,
    componentClientId,
  };
};

export const getMatchingAccountId = memoizeOne((clientConfigs, clientId, appId) => {
  const matchingClient = clientConfigs.find((cl) => cl.clientId === clientId) || {};
  const { apps = [] } = matchingClient;
  const matchingApp = apps.find((app) => app.appId === appId) || {};
  const { accountId = 'default' } = matchingApp;
  return accountId;
});

export const getComponentClientIdForClient = memoizeOne((clientConfigs, clientId) => {
  const matchingClient = clientConfigs.find((cl) => cl.clientId === clientId) || {};
  return matchingClient.componentClientId;
});

export const isUserAuthorized = (userSelectedCAA = {}, clientsList = [{}]) => {
  try {
    const {
      clientId: userClientId,
      appId: userAppId,
      accountId: userAccountId
    } = userSelectedCAA;
    // caa information is neither present in URL or cookie
    // user is authorized to view default CAA
    if (userClientId === undefined || userClientId === '') {
      if (clientsList.length === 1 && clientsList[0].apps.length === 1) {
        return {
          authorized: true,
          clientPicker: false,
          caa: getDefaultCAA(clientsList)
        };
      }
      return {
        authorized: true,
        clientPicker: true,
        caa: {}
      };
    }

    // caa information is present in URL.
    const clientConfig = clientsList.find((client) => client.clientId === userClientId);
    if (clientConfig) {
      if (userAppId === undefined && userAccountId === '') {
        if (clientConfig.apps.length === 1) {
          return {
            authorized: true,
            clientPicker: false,
            caa: {
              clientId: userClientId,
              appId: clientConfig.apps[0].appId,
              accountId: clientConfig.apps[0].accountId,
              componentClientId: clientConfig.componentClientId,
            }
          };
        }
        return {
          authorized: true,
          clientPicker: true,
          caa: {}
        };
      }

      let appDetails = clientConfig.apps.find((app) => app.appId === userAppId);
      if (!appDetails) {
        const [first] = clientConfig.apps;
        if (first) {
          appDetails = first;
        }
      }
      // we were able to find matching client and app.
      const { accountId, appId } = appDetails;
      userSelectedCAA.appId = appId;
      userSelectedCAA.accountId = accountId;
      userSelectedCAA.componentClientId = clientConfig.componentClientId;

      return {
        authorized: true,
        clientPicker: false,
        caa: userSelectedCAA
      };
    }
  } catch (err) {
    devlog.warn(err);
  }
  // return false and defaultCAA if we were not able to find matching CAA in clientsList
  return {
    authorized: false,
    clientPicker: false,
    caa: getDefaultCAA(clientsList)
  };
};
