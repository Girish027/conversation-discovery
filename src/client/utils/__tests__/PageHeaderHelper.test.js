import * as PageHeaderHelper from 'utils/PageHeaderHelper';
import * as routerActions from 'connected-react-router';
import { getRoute, routeNames } from 'utils/RouteHelper';

jest.mock('connected-react-router');

describe('utils/PageHeaderHelper', function () {
  const commonData = {
    itsBaseUrl: 'https://its.com/',
    clientId: '247ai',
    accountId: 'tfscorp',
    appId: 'referencebot',
    dispatch: () => {},
  };
  afterAll(() => {
    jest.clearAllMocks();
  });

  describe('detailsMap', () => {
    test('should have mappings between route, title and breadcrumbs', () => {
      expect(PageHeaderHelper.detailsMap).toMatchSnapshot();
    });
  });

  describe('getPageDetails', () => {
    beforeAll(function () {

    });
    test('should get correct title and breadcrumbs for Discover Intents Route', () => {
      const pageData = {
        projectId: 'pro-123',
        projectName: 'project 123',
        routeName: routeNames.DISCOVER_INTENTS
      };
      expect(PageHeaderHelper.getPageDetails({ ...commonData, ...pageData })).toMatchSnapshot();
    });

    test('should get correct title and breadcrumbs for Distribution Graph Route', () => {
      const pageData = {
        projectId: 'pro-123',
        projectName: 'project 123',
        runId: 'run-123',
        runName: 'Run 123',
        routeName: routeNames.DISTRIBUTION_GRAPH
      };
      expect(PageHeaderHelper.getPageDetails({ ...commonData, ...pageData })).toMatchSnapshot();
    });

    test('should get correct title and breadcrumbs for Topic Review Route', () => {
      const pageData = {
        projectId: 'pro-123',
        projectName: 'project 123',
        runId: 'run-123',
        runName: 'Run 123',
        clusterId: 'cluster-234',
        clusterName: 'cancel_order',
        routeName: routeNames.TOPIC_REVIEW
      };
      expect(PageHeaderHelper.getPageDetails({ ...commonData, ...pageData })).toMatchSnapshot();
    });
  });

  describe('clickHandlers', () => {
    describe('goToBotOverview', () => {
      beforeEach(function () {
        window.location.assign = jest.fn();
      });
      test('should navigate to given itsURL', () => {
        const data = commonData;
        expect(PageHeaderHelper.goToBotOverview(data));
        expect(window.location.assign).not.toHaveBeenCalledWith(`${data.itsBaseUrl}?clientid=${data.clientId.toLowerCase()}&accountid=${data.accountId.toLowerCase()}&appid=${data.appId.toLowerCase()}`);
        expect(window.location.assign).toHaveBeenCalledWith('#');
      });
    });

    describe('Specific handler - should change route', () => {
      const pageData = {
        projectId: 'pro-123',
        projectName: 'project 123',
        runId: 'run-123',
        runName: 'Run 123',
        clusterId: 'cluster-234',
        clusterName: 'cancel_order',
        routeName: `${routeNames.DISTRIBUTION_GRAPH}/`, // this is how the route location would be
      };
      let pageDetails;

      beforeEach(function () {
        commonData.dispatch = jest.fn();
        routerActions.push = jest.fn();
        pageDetails = PageHeaderHelper.getPageDetails({ ...commonData, ...pageData });
      });

      test('should navigate to corresponding route when clicked on project name', () => {
        routerActions.push.mockReturnValue('router action push');
        const { breadcrumb } = pageDetails;
        const projectBreadcrumb = breadcrumb[1];
        const { onClick: onClickProject } = projectBreadcrumb;
        onClickProject();
        expect(commonData.dispatch).toMatchSnapshot();
        expect(routerActions.push).toHaveBeenCalledWith(getRoute(routeNames.DISCOVER_INTENTS, { ...commonData, ...pageData }));
      });

    });
  });
});
