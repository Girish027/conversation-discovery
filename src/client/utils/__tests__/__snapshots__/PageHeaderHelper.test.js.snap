// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/PageHeaderHelper clickHandlers Specific handler - should change route should navigate to corresponding route when clicked on project name 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "router action push",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`utils/PageHeaderHelper detailsMap should have mappings between route, title and breadcrumbs 1`] = `
Object {
  "/conversationdiscovery": Object {
    "title": [Function],
  },
  "/conversationdiscovery/ui": Object {
    "title": [Function],
  },
  "/conversationdiscovery/ui/discover-intents": Object {
    "title": [Function],
  },
  "/conversationdiscovery/ui/run-overview": Object {
    "breadcrumbItems": Array [
      Object {
        "label": "discoverIntents",
      },
      Object {
        "label": "projectView",
        "route": "/conversationdiscovery/ui/discover-intents",
      },
    ],
    "title": [Function],
  },
  "/conversationdiscovery/ui/topic-review": Object {
    "breadcrumbItems": Array [
      Object {
        "label": "discoverIntents",
      },
      Object {
        "label": "projectView",
        "route": "/conversationdiscovery/ui/discover-intents",
      },
      Object {
        "label": "runView",
        "route": "/conversationdiscovery/ui/run-overview",
      },
    ],
    "title": [Function],
  },
}
`;

exports[`utils/PageHeaderHelper getPageDetails should get correct title and breadcrumbs for Discover Intents Route 1`] = `
Object {
  "breadcrumb": Array [],
  "title": "Discover Intents",
}
`;

exports[`utils/PageHeaderHelper getPageDetails should get correct title and breadcrumbs for Distribution Graph Route 1`] = `
Object {
  "breadcrumb": Array [
    Object {
      "label": "Discover Intents",
      "onClick": undefined,
      "readOnly": true,
      "value": "Discover Intents",
    },
    Object {
      "accountId": "tfscorp",
      "appId": "referencebot",
      "clientId": "247ai",
      "itsBaseUrl": "https://its.com/",
      "label": "project 123",
      "onClick": [Function],
      "projectId": "pro-123",
      "projectName": "project 123",
      "readOnly": false,
      "routeName": "/conversationdiscovery/ui/run-overview",
      "runId": "run-123",
      "runName": "Run 123",
      "value": "projectproject 123",
    },
  ],
  "title": "Run 123 Overview",
}
`;

exports[`utils/PageHeaderHelper getPageDetails should get correct title and breadcrumbs for Topic Review Route 1`] = `
Object {
  "breadcrumb": Array [
    Object {
      "label": "Discover Intents",
      "onClick": undefined,
      "readOnly": true,
      "value": "Discover Intents",
    },
    Object {
      "accountId": "tfscorp",
      "appId": "referencebot",
      "clientId": "247ai",
      "clusterId": "cluster-234",
      "clusterName": "cancel_order",
      "itsBaseUrl": "https://its.com/",
      "label": "project 123",
      "onClick": [Function],
      "projectId": "pro-123",
      "projectName": "project 123",
      "readOnly": false,
      "routeName": "/conversationdiscovery/ui/topic-review",
      "runId": "run-123",
      "runName": "Run 123",
      "value": "projectproject 123",
    },
    Object {
      "accountId": "tfscorp",
      "appId": "referencebot",
      "clientId": "247ai",
      "clusterId": "cluster-234",
      "clusterName": "cancel_order",
      "itsBaseUrl": "https://its.com/",
      "label": "Run 123 Overview",
      "onClick": [Function],
      "projectId": "pro-123",
      "projectName": "project 123",
      "readOnly": false,
      "routeName": "/conversationdiscovery/ui/topic-review",
      "runId": "run-123",
      "runName": "Run 123",
      "value": "runRun 123 Overview",
    },
  ],
  "title": "Topic Review",
}
`;
